internal/common/resource.go

```diff
@@ -431,3 +432,108 @@ func ResourceDeletedResult(resource client.Object, res OperationResult) Reconcil
 		OperationResult: res,
 	}
 }
+
+func defaultUpdateFunc(newObj, foundObj client.Object) {
+	switch newTyped := newObj.(type) {
+	case *core.ConfigMap:
+		foundConfigMap := foundObj.(*core.ConfigMap)
+		foundConfigMap.Immutable = newTyped.Immutable
+		foundConfigMap.Data = newTyped.Data
+		foundConfigMap.BinaryData = newTyped.BinaryData
+
+	case *core.Namespace:
+		// Intentionally empty
+
+	case *core.Service:
+		foundService := foundObj.(*core.Service)
+		// ClusterIP should not be updated
```

Comment:
What does this mean?

Reply:
The `ClusterIP` field is set be the API server and is immutable. The problem is that `newObj` does not have the `ClusterIP` set, so if we would simply set the sepc:
```go
foundObj.spec = newObj.spec
```
it would set `ClusterIP` to `nil`, and the update would fail.

Reply:
ACK